test/checkout.int.t.sol

```diff
@@ -962,26 +964,28 @@ contract Checkout_buyingWithAlternateERC20_floatingPointPriceMultiple is
   }
 }
 
-contract Checkout_buyingWithCustomFee is Checkout {
+contract Checkout_swapWithoutFeeSpecialOrder is Checkout {
   uint256 ownerPrivateKey = 0xA11CE;
   address owner = vm.addr(ownerPrivateKey);
   uint256 customFee = 5;
   uint256 certificateAmount = 1 ether;
+  uint256 customPriceMultiple = 1800; // $18.00 -- test below the default price multiple of $20.00
 
   function setUp() external {
     _removalIds = _seedRemovals({
       to: _namedAccounts.supplier,
       count: 1,
       list: true
     });
-    uint256 purchaseAmount = _market.calculateCheckoutTotalWithoutFee(
-      certificateAmount
-    );
+    uint256 purchaseAmount = _market.calculateCheckoutTotalWithoutFee({
+      amount: certificateAmount,
+      priceMultiple: customPriceMultiple
+    });
     _market.grantRole({role: _market.MARKET_ADMIN_ROLE(), account: owner});
     vm.prank(_namedAccounts.admin);
     _bpNori.deposit(owner, abi.encode(purchaseAmount));
     vm.prank(owner);
-    _bpNori.approve(address(_market), MAX_INT); // infinite approval for Market to spend owner's tokens
+    _bpNori.approve(address(_market), purchaseAmount);
```

Comment:
I wanted to tighten up this logic and discovered that when the approval amount is `MAX_INT` then the approval balance isn't updated upon token transfer.  this means that in the case that the approval amount is `MAX_INT` there are two fewer events that are emitted in the flow of one of our purchases, because two `Approval` events that otherwise would have been emitted (one on transfer to the supplier and one on transfer to the rNORI contract.... and in a standard swap scenario a THIRD approval event upon transfer to the fee wallet) do not get emitted.  this changes the index of the event signatures we care about.  this one was a bit of a stumper for a while so figured I'd share here